overlay/master_server.py

```diff
@@ -280,21 +224,39 @@ def SubmitValidationJob(self, request, context):
             worker_job.gis_joins.append(gis_join)
             job.worker_jobs[worker.hostname] = worker_job
 
+        # Log selected workers
         for worker_hostname, worker_job in job.worker_jobs.items():
             info(f"{worker_hostname}: {worker_job}")
 
-        if request.job_mode == "multithreaded":
-            info("Launching jobs in multithreaded mode")
+        # Validation budget inference
+        if request.validation_budget.budget_type == BudgetType.STATIC_BUDGET:
+            static_budget: ValidationBudget = request.validation_budget.static_budget
+            if static_budget.total_limit > 0:
+                if static_budget.total_limit > len(request.gis_joins):
+                    static_budget.strata_limit = static_budget.total_limit // len(request.gis_joins)
+                else:
+                    info("Specified a total limit less than the number of GISJOINs. Defaulting to 1 per GISJOIN")
+                    static_budget.strata_limit = 1
+        elif request.validation_budget.budget_type == BudgetType.INCREMENTAL_VARIANCE_BUDGET:
+            error("Incremental variance budget not yet implemented!")
+            return ValidationJobResponse(id=job_id, metrics=[])
+        else:
+            error(f"Unknown budget type {request.validation_budget.budget_type}!")
+            return ValidationJobResponse(id=job_id, metrics=[])
+
+        # Select strategy for submitting the job from the master
+        if request.master_job_mode == JobMode.MULTITHREADED:
+            info("Launching jobs in multi-threaded mode")
             validation_job_responses = launch_worker_jobs_multithreaded(job, request)
-        elif request.job_mode == "asynchronous":
+        elif request.master_job_mode == JobMode.ASYNCHRONOUS:
```

Comment:
I used `enum` instead of string comparison and added `master_job_mode` and `worker_job_mode` instead of a single `job_mode`